src/main/java/com/google/edith/servlets/UserInsights.java

```diff
@@ -0,0 +1,194 @@
+package com.google.edith.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * This class provides the funttionality to parse specifc user data from 
+ * datastore and send an agreggate of that information in a JSON file.
+ */
+public final class UserInsights {
+
+  private String userId;
+
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();  
+  private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+  private static final Gson gson = new Gson();
+
+  /** Assumes dates are in the form yyyy-MM-dd */
+  private static final Comparator<Key> SORT_BY_DATE = (Key item1, Key item2) -> {
+      try {
+        return ((String) (datastore.get(item1).getProperty("date")))
+            .compareTo((String) (datastore.get(item2).getProperty("date")));
+      } catch(EntityNotFoundException | NullPointerException e )  {
+        return 0;
+      }
+  };
+  
+  public UserInsights(String userId) {
+    this.userId = userId;
```

Comment:
I was wondering that too. I could remove it, then make the rest of the methods static. I created this constructor in case we wanted to pass in more information, but so far userId was the only variable I could think of that would be unique for each object.